snuba/utils/streams/kafka.py

```diff
@@ -34,7 +36,42 @@ class TransportError(ConsumerError):
     pass
 
 
+KafkaConsumerState = Enum("KafkaConsumerState", ["CONSUMING", "ERROR", "CLOSED"])
```

Comment:
Does mypy work properly with this way of defining enums compared to 

```
KafkaConsumerState(Enum):
   CONSUMING =1
   ....
```

Reply:
Yep:

```diff
diff --git a/snuba/utils/streams/kafka.py b/snuba/utils/streams/kafka.py
index 8775eba..d27d0ca 100644
--- a/snuba/utils/streams/kafka.py
+++ b/snuba/utils/streams/kafka.py
@@ -38,6 +38,8 @@ class TransportError(ConsumerError):
 
 KafkaConsumerState = Enum("KafkaConsumerState", ["CONSUMING", "ERROR", "CLOSED"])
 
+reveal_type(KafkaConsumerState.CONSUMING)
+
 
 class InvalidState(RuntimeError):
     def __init__(self, state: KafkaConsumerState):
@@ -105,6 +107,7 @@ class KafkaConsumer(Consumer[TopicPartition, int, bytes]):
         )
 
         self.__state = KafkaConsumerState.CONSUMING
+        reveal_type(self.__state)
 
     def __resolve_partition_offset_earliest(
         self, partition: ConfluentTopicPartition
```

```
snuba/utils/streams/kafka.py:41: note: Revealed type is 'snuba.utils.streams.kafka.KafkaConsumerState*'
snuba/utils/streams/kafka.py:110: note: Revealed type is 'snuba.utils.streams.kafka.KafkaConsumerState*'
```